### PR TITLE
Backports 0.18 pr20

### PR DESCRIPTION
**Pull request by Munkybooty**
I fixed as much as i could in terms of test failures, but not everything passes on the functional tests

### CODE REVIEW

File: test/functional/feature_block.py [hunk 1]
```diff
@@ -1269,20 +1269,32 @@ def run_test(self):
...
+        b_v1 = self.next_block('b_v1', version=1)
+        self.sync_blocks([b_v1], success=False, force_send=True, reject_reason='bad-version(0x00000001)')
+
+        self.move_tip(chain1_tip + 2)
+        b_cb34 = self.next_block('b_cb34', version=4)
+        b_cb34.vtx[0].vin[0].scriptSig = b_cb34.vtx[0].vin[0].scriptSig[:-1]
+        b_cb34.vtx[0].rehash()
+        b_cb34.hashMerkleRoot = b_cb34.calc_merkle_root()
+        b_cb34.solve()
+        self.sync_blocks([b_cb34], success=False, reject_reason='bad-cb-height', reconnect=True)
```
Comment (PastaPastaPasta) [inline test/functional/feature_block.py:1296]: This is missing some previous back port to allow us to use success and reject_reason

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -4030,7 +4030,7 @@ UniValue walletcreatefundedpsbt(const JSONRPCRequest& request)
         return NullUniValue;
     }
 
-    if (request.fHelp || request.params.size() < 2 || request.params.size() > 6)
+    if (request.fHelp || request.params.size() < 2 || request.params.size() > 5)
```
[the remark this answers is not in the document]
Comment (Munkybooty) [reply src/wallet/rpcwallet.cpp:4033]: iirc, the remainder of the pr deals entirely with rbf

File: test/functional/feature_block.py (hunk 1 above)
Comment (PastaPastaPasta) [reply test/functional/feature_block.py:1296]: did you see this?

Comment (Munkybooty) [reply test/functional/feature_block.py:1296]: Oh shoot, I'm seeing this now. Will get on it

Comment (Munkybooty) [reply test/functional/feature_block.py:1296]: Are you sure its 11817, it only touches `feature_csv_activation.py`

**Comment by PastaPastaPasta** [comment]
c++20 build fails + tests fail

**Comment by Munkybooty** [comment]
c++20 build fixed by rebase, need to still investigate these test failures

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see below + https://github.com/UdjinM6/dash/commits/pr4675

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
LGTM, pls drop all commits related to 14738 (3b6848a, f82c23d, d4bafcb, cc86ef8, 71e16e2, 9d625c7) and it should be ready to merge imo

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**